### PR TITLE
SDK-522 - SDK Side adjustment to support SPARKZ Changes

### PR DESCRIPTION
**Pull request by i-skrypnyk**

### CODE REVIEW

File: evm/pom.xml [hunk 1]
```diff
@@ -55,9 +55,9 @@
       <version>2.19.0</version>
     </dependency>
     <dependency>
-      <groupId>org.scorexfoundation</groupId>
+      <groupId>io.horizen</groupId>
       <artifactId>scrypto_2.12</artifactId>
-      <version>2.1.7</version>
+      <version>2.0.0-RC10-SNAPSHOT-LOCAL</version>
```
Comment (i-Alex) [inline evm/pom.xml:60]:
```suggestion
      <version>2.0.0-RC10-SNAPSHOT</version>
```

**Review by i-Alex — CHANGES_REQUESTED** [review]

File: evm/pom.xml (hunk 1 above)
Comment (i-skrypnyk) [reply evm/pom.xml]: thanks, fixed

File: qa/resources/template.conf [hunk 2]
```diff
@@ -18,8 +18,9 @@ sparkz {
     bindAddress = "%(API_ADDRESS)s:%(BIND_PORT)s"
     knownPeers = []
     agentName = "2-Hop"
-    maxConnections = %(MAX_CONNECTIONS)d
-    maxPacketSize = %(MAX_PACKET_SIZE)d
+    maxIncomingConnections = %(MAX_INCOMING_CONNECTIONS)d
+    maxOutgoingConnections = %(MAX_OUTGOING_CONNECTIONS)d
+    maxModifiersSpecMessageSize = %(MAX_PACKET_SIZE)d
```
Comment (i-Alex) [inline qa/resources/template.conf:23]: `GET_PEERS_INTERVAL` is missing in this file.
So the value specified in `initialize_sc_datadir` is never used.
same for `qa/resources/template_predefined_genesis.conf` file

**Review by i-Alex — CHANGES_REQUESTED** [review]
Looks good
Minor point about the STF and an issue with sparkz dependencies not published

File: qa/resources/template.conf [hunk 3]
```diff
@@ -18,8 +18,10 @@ sparkz {
     bindAddress = "%(API_ADDRESS)s:%(BIND_PORT)s"
     knownPeers = []
     agentName = "2-Hop"
-    maxConnections = %(MAX_CONNECTIONS)d
-    maxPacketSize = %(MAX_PACKET_SIZE)d
+    getPeersInterval = %(GET_PEERS_INTERVAL)d
```
Comment (paologalligit) [inline qa/resources/template.conf:21]: python tests are failing with the error `Unexpected exception caught during testing: %d format: a number is required, not str`. Changed at the end with s -> `getPeersInterval = %(GET_PEERS_INTERVAL)s`

Comment (i-skrypnyk) [reply qa/resources/template.conf:21]: ahh, sorry, fixed it

**Review by i-Alex — APPROVED** [review]

**Review by paologalligit — APPROVED** [review]

**Review by i-Alex — APPROVED** [review]

**Pull request merged**